packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_time.dart

```diff
@@ -0,0 +1,128 @@
+import 'temporal.dart';
+
+///
+class TemporalTime {
+  DateTime _dateTime;
+  int _nanoseconds; // DateTime only stores millisecond and microsecond
+  Duration _offset;
+  bool _hasZ =
+      false; // fromString only - Need to track between hh:mm:ss vs hh:mm:ssZ
+
+  ///
+  static TemporalTime now() {
+    return TemporalTime(DateTime.now());
+  }
+
+  ///
+  TemporalTime(DateTime dateTime) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(
+        1970, 1, 1, dateTime.hour, dateTime.minute, dateTime.second);
+    _nanoseconds =
+        (dateTime.millisecond * 1000000) + (dateTime.microsecond * 1000);
+  }
+
+  ///
+  TemporalTime.withOffset(DateTime dateTime, Duration offset) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(
+        1970, 1, 1, dateTime.hour, dateTime.minute, dateTime.second);
+    _nanoseconds =
+        (dateTime.millisecond * 1000000) + (dateTime.microsecond * 1000);
+
+    if (offset.inDays > 0) {
+      throw new Exception("Cannot have an offset in days (hh:mm:ss)");
+    }
+
+    _offset = offset;
+  }
+
+  //hh:mm
+  //hh:mmZ
+  //hh:mm:ss
+  //hh:mm:ssZ
+  //hh:mm:ss.sss
+  //hh:mm:ss.sssZ
+  // without Z:
+  //   +hh:mm
+  //   +hh:mm:ss
+  TemporalTime.fromString(String iso8601String) {
+    RegExp regExp = new RegExp(
+        r'^([0-2][0-9]):([0-5][0-9])(:([0-5][0-9])(.([0-9]{1,9}))?)?((z|Z)|((\+|-)[0-2][0-9]:[0-5][0-9](:[0-5][0-9])?))?',
+        caseSensitive: false,
+        multiLine: false);
+
+    // Validate
+    String regexString = regExp.stringMatch(iso8601String);
+    if (regexString != iso8601String) {
+      throw new Exception("invalid string input");
+    }
+
+    // Remove Z
+    regexString = iso8601String.replaceAll(new RegExp(r'(z|Z)'), '');
+
+    // Extract Time
+    var match = regExp.matchAsPrefix(regexString);
+
+    int hours = int.parse(match.group(1));
+    int minutes = int.parse(match.group(2));
+    int seconds = Temporal.getIntOr0(match.group(4));
+    _nanoseconds = Temporal.getIntOr0(match.group(6));
+    //int milliseconds =
+
+    _dateTime = DateTime.utc(1970, 1, 1, hours, minutes, seconds);
```

Comment:
The reason is to be consistent with Android's implementation which sets it to 1970/01/01

This can impact developers as the method "toDateTime" returns a Dart DateTime which needs to have 1970/01/01 (Android has something similar in their toDate method which auto sets to 1970/01/01)